tests/Unit/NumericalAlgorithms/LinearOperators/Test_PartialDerivatives.cpp

```diff
@@ -368,3 +371,162 @@ SPECTRE_TEST_CASE("Unit.Numerical.LinearOperators.PartialDerivs",
+
+  const auto& du = db::get<deriv_tag>(box);
+
+  tmpl::for_each<vars_tags>([&du, &expected_du, &extents ](auto tag) noexcept {
+    using Tag = tmpl::type_from<decltype(tag)>;
+    using DerivativeTag = Tags::deriv<Tag, tmpl::size_t<Dim>, Frame::Logical>;
+    auto& expected_dvariable = get<DerivativeTag>(expected_du);
+    for (auto it = expected_dvariable.begin(); it != expected_dvariable.end();
+         ++it) {
+      const auto deriv_indices = expected_dvariable.get_tensor_index(it);
+      const size_t deriv_index = deriv_indices[0];
+      const auto tensor_indices =
+          all_but_specified_element_of<0>(deriv_indices);
+      for (size_t n = 0; n < extents.product(); ++n) {
+        CAPTURE_PRECISE(get<Tag>(du[deriv_index]).get(tensor_indices)[n] -
+                        (*it)[n]);
+        CHECK(get<Tag>(du[deriv_index]).get(tensor_indices)[n] ==
+              approx((*it)[n]));
+      }
+    }
+  });
+}
+
+template <class MapType>
+struct MapTag : db::DataBoxTag {
+  using type = MapType;
+  static constexpr db::DataBoxString label = "MapTag";
+};
+
+template <size_t Dim, typename T>
+void test_partial_derivatives_compute_item(
+    const std::array<size_t, Dim> extents_array, const T& map) noexcept {
+  using vars_tags = tmpl::list<Var1<Dim>, Var2>;
+  using map_tag = MapTag<std::decay_t<decltype(map)>>;
+  using inv_jac_tag =
+      Tags::InverseJacobian<map_tag, Tags::LogicalCoordinates<Dim>>;
+  using deriv_tag = Tags::deriv<vars_tags, vars_tags, inv_jac_tag>;
+
+  const std::array<size_t, Dim> array_to_functions{extents_array -
+                                                   make_array<Dim>(size_t{1})};
+  const Index<Dim> extents{extents_array};
+  Variables<vars_tags> u(extents.product());
+  Variables<
+      db::wrap_tags_in<Tags::deriv, vars_tags, tmpl::size_t<Dim>, Frame::Grid>>
+      expected_du(extents.product());
+  const auto x_logical = logical_coordinates(extents);
+  const auto x = map(logical_coordinates(extents));
+
+  tmpl::for_each<vars_tags>([&array_to_functions, &x, &u ](auto tag) noexcept {
+    using Tag = tmpl::type_from<decltype(tag)>;
+    get<Tag>(u) = Tag::f(array_to_functions, x);
+  });
+  tmpl::for_each<vars_tags>(
+      [&array_to_functions, &x, &expected_du ](auto tag) noexcept {
+        using Tag = typename decltype(tag)::type;
+        using DerivativeTag = Tags::deriv<Tag, tmpl::size_t<Dim>, Frame::Grid>;
+        get<DerivativeTag>(expected_du) = Tag::df(array_to_functions, x);
+      });
+
+  auto box =
+      db::create<db::AddTags<Tags::Extents<Dim>, Tags::Variables<vars_tags>,
+                             Tags::LogicalCoordinates<Dim>, map_tag>,
+                 db::AddComputeItemsTags<inv_jac_tag, deriv_tag>>(
+          extents, u, x_logical, map);
+
+  const auto& du = db::get<deriv_tag>(box);
+
+  for (size_t n = 0; n < du.size(); ++n) {
+    CAPTURE_PRECISE(du.data()[n] - expected_du.data()[n]);  // NOLINT
```

Comment:
Explain NOLINT

Reply:
@kidder why is this NOLINT? I stole it from what you wrote :stuck_out_tongue:

Reply:
pointer arithmetic